### DISCUSSION TITLE
Complaining about missing DLL when attempt to run wxWidgets minimal application

### DISCUSSION THREAD

**Discussion by maurobio** [PBfordev/wxpbguide#7] [General]
Hi, sorry for bothering, but I used your truly great guide for installing wxWidgets under Ms-Windows and found an annoying problem: I carefully followed the steps outlined in the guide to: (1) install the MSYS2 system; (2) download and compile the latest version of wxWidgets (3.2.6) in both "Debug" and "Release" versions, using the batch file "build-library-dhared.bat" you have provided. Everything ran just fine and without any hassles so far. 

I then proceeded to compile the wxWidgets minimal application, again using the batch file you have provided ("build-sample-minimal-shared-debug.bat"). Compilation was successful, and then I copied the required  files (as per your guide) to the same directory of the minimal.exe application. But a dialog box keeps being displayed, complaining about a missing "wxmsw32u_core_gcc_custom.dll", but it is still there, along with the other ones!

This is most frustrating! Could you please give me some hints on how this could be solved?

Thank you very much in advance, and please accept my sincere congratulations for your great wxWidgets installation guide.

With warmest regards,
Mauro

**Comment by maurobio** [comment]
Hi, here is an update: I finally got the minimal application working, by following the suggestion from here: [https://forums.wxwidgets.org/viewtopic.php?t=42302](https://forums.wxwidgets.org/viewtopic.php?t=42302) to add an entry: "C:\wxWidgets-3..2.6\lib\gcc_dll" to my PATH variable. This is a workaround, but seems sensible enough, as it avoid having to copy the required DLL's to all and every directory of any application created with wxWidgets.

Best regards,
Mauro

**Comment by PBfordev** [comment]
> I then proceeded to compile the wxWidgets minimal application, again using the batch file you have provided ("build-sample-minimal-shared-debug.bat"). Compilation was successful, and then I copied the required files (as per your guide) to the same directory of the minimal.exe application. But a dialog box keeps being displayed, complaining about a missing "wxmsw32u_core_gcc_custom.dll", but it is still there, along with the other ones!

I can't think of any way this could happen: You built the sample in the debug configuration, yet it seems linked to non-debug configuration wxWidgets library (notice the missing "d" in the library name in your error message, the debug version is named wxmsw32u**d**_core_gcc_custom.dll)?!

My guide explains why adding a folder with a DLL library to PATH is not the best idea (a possibility of an incompatible library being there), but as long as it keeps working for you...

**Comment by maurobio** [comment]
Hi! Thanks for the fast reply! Indeed, it was my mistake, sorry. I was just
copying the wrong version of the dll to the minimal.exe application
directory.

But I found another problem, unrelated to that one: after successfully
installing Code::Blocks and following the steps outlined in your guide for
configuring it, I noticed that the option "Have g++ follow the C++20 ISO
C++ language standard (-std=c++20)" did not appear in the list displayed in
Compiler settings -> Compiler flags. In fact, the only options for C++
which appeared for me were the 1998 standard (see attached screenshot)!.
But this was easy enough to fix, by just creating a flag with the required
compiler directive (-std=c++20). After that, everything ran smoothly and I
have been able to create, compile, and run the test application without
further problems.

With best regards,
Mauro

Em qui., 28 de nov. de 2024 às 05:00, PB ***@***.***>
escreveu:

> I then proceeded to compile the wxWidgets minimal application, again using
> the batch file you have provided ("build-sample-minimal-shared-debug.bat").
> Compilation was successful, and then I copied the required files (as per
> your guide) to the same directory of the minimal.exe application. But a
> dialog box keeps being displayed, complaining about a missing
> "wxmsw32u_core_gcc_custom.dll", but it is still there, along with the other
> ones!
>
> I can't think of any way this could happen: You built the sample in the
> debug configuration, yet it seems linked to non-debug configuration
> wxWidgets library (notice the missing "d" in the library name in your error
> message, the debug version is named wxmsw32u*d*_core_gcc_custom.dll)?!
>
> My guide explains why adding a folder with a DLL library to PATH is not
> the best idea (a possibility of an incompatible library being there), but
> as long as it keeps working for you...
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/PBfordev/wxpbguide/discussions/7#discussioncomment-11403267>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAPMACNHHZV77ZDI4SD3YVL2C3ESNAVCNFSM6AAAAABSTVC4COVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNBQGMZDMNY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Dr. Mauro J. Cavalcanti
E-mail: ***@***.***
GitHub: https://github.com/maurobio
ORCID: https://orcid.org/0000-0003-2389-1902
"Life is complex. It consists of real and imaginary parts."

**Comment by PBfordev** [comment]
> Hi! Thanks for the fast reply! Indeed, it was my mistake, sorry. 

I'm glad to hear the issue has been solved so the Discussion can be closed.

> Have g++ follow the C++20 ISO C++ language standard (-std=c++20)" did not appear in the list displayed in Compiler settings -> Compiler flags.

I think a bunch of compiler options sometimes disappearing from the list is a known C::B issue. I had encountered it once, but I don't remember if I just needed to restart C::B or had to do something to fix it.
